packages/prerender/src/runPrerender.tsx

```diff
@@ -10,59 +10,82 @@ import { registerApiSideBabelHook } from '@redwoodjs/internal/dist/build/babel/a
+import { GqlHandlerImportError, PrerenderGqlError } from './errors'
 import { executeQuery, getGqlHandler } from './graphql/graphql'
 import { getRootHtmlPath, registerShims, writeToDist } from './internal'
 
-export class PrerenderGqlError {
-  message: string
-  stack: string
-
-  constructor(message: string) {
-    this.message = 'GQL error: ' + message
-    // The stacktrace would just point to this file, which isn't helpful,
-    // because that's not where the error is. So we're just putting the
-    // message there as well
-    this.stack = this.message
-  }
-}
-
 async function recursivelyRender(
   App: React.ElementType,
   renderPath: string,
   gqlHandler: any,
   queryCache: Record<string, QueryInfo>
 ): Promise<string> {
+  let shouldShowGraphqlHandlerNotFoundWarn = false
   // Execute all gql queries we haven't already fetched
   await Promise.all(
     Object.entries(queryCache).map(async ([cacheKey, value]) => {
       if (value.hasFetched) {
-        // Already fetched this one; skip it!
+        // Already fetched, or decided that we can't render this one; skip it!
         return Promise.resolve('')
       }
 
-      const resultString = await executeQuery(
-        gqlHandler,
-        value.query,
-        value.variables
-      )
-      const result = JSON.parse(resultString)
-
-      if (result.errors) {
-        const message =
-          result.errors[0].message ?? JSON.stringify(result.errors)
-        throw new PrerenderGqlError(message)
-      }
-
-      queryCache[cacheKey] = {
-        ...value,
-        data: result.data,
-        hasFetched: true,
+      try {
+        const resultString = await executeQuery(
+          gqlHandler,
+          value.query,
+          value.variables
+        )
+
+        const result = JSON.parse(resultString)
+
+        if (result.errors) {
+          const message =
+            result.errors[0].message ?? JSON.stringify(result.errors, null, 4)
+
+          if (result.errors[0]?.extensions?.code === 'UNAUTHENTICATED') {
+            console.error(
+              `\n \n 🛑  Cannot prerender the query ${getOperationName(
+                value.query
+              )} as it requires auth. \n`
+            )
+          }
+
+          throw new PrerenderGqlError(message)
+        }
+
+        queryCache[cacheKey] = {
+          ...value,
+          data: result.data,
+          hasFetched: true,
+        }
+
+        return result
+      } catch (e) {
+        if (e instanceof GqlHandlerImportError) {
+          // We need to need to swallow the error here, so that
+          // we can continue to render the page, with cells in loading state
+          // e.g. If the GQL handler is located elsewhere
+          shouldShowGraphqlHandlerNotFoundWarn = true
+
+          queryCache[cacheKey] = {
+            ...value,
+            renderLoading: true,
+            hasFetched: true, // tried to fetch, but failed
```

Comment:
Maybe rename the variable to `skipFetch` now that it signals more things than it did before?

Reply:
I think the combination of "renderLoading" + "hasFetched" === "Skip cell prerendering"

The reason I didn't want to reuse the same variable was because:
a) The recursion hinges on "hasFetched" - i.e. hasFetched is the base condition - skipFetch wouldn't describe the same case.
b) It's a little confusing when we render the loader, and when we error out. Having the "renderLoading" in there makes it explicit.